charts/internal/shoot-system-components/charts/metallb/templates/metallb.yaml

```diff
@@ -0,0 +1,419 @@
+  name: pod-lister
+  namespace: metallb-system
+rules:
+- apiGroups:
+  - ''
+  resources:
+  - pods
+  verbs:
+  - list
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  labels:
+    app: metallb
+  name: metallb-system:controller
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: metallb-system:controller
+subjects:
+- kind: ServiceAccount
+  name: controller
+  namespace: metallb-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  labels:
+    app: metallb
+  name: metallb-system:speaker
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: metallb-system:speaker
+subjects:
+- kind: ServiceAccount
+  name: speaker
+  namespace: metallb-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  labels:
+    app: metallb
+  name: config-watcher
+  namespace: metallb-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: config-watcher
+subjects:
+- kind: ServiceAccount
+  name: controller
+- kind: ServiceAccount
+  name: speaker
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  labels:
+    app: metallb
+  name: pod-lister
+  namespace: metallb-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: pod-lister
+subjects:
+- kind: ServiceAccount
+  name: speaker
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  labels:
+    app: metallb
+    component: speaker
+  name: speaker
+  namespace: metallb-system
+spec:
+  selector:
+    matchLabels:
+      app: metallb
+      component: speaker
+  template:
+    metadata:
+      annotations:
+        prometheus.io/port: '7472'
+        prometheus.io/scrape: 'true'
+      labels:
+        app: metallb
+        component: speaker
+    spec:
+      containers:
+      - args:
+        - --port=7472
+        - --config=config
+        env:
+        - name: METALLB_NODE_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
+        - name: METALLB_HOST
+          valueFrom:
+            fieldRef:
+              fieldPath: status.hostIP
+        # Disable memberlist because we're using BGP.
+        # - name: METALLB_ML_BIND_ADDR
+        #   valueFrom:
+        #     fieldRef:
+        #       fieldPath: status.podIP
+        # needed when another software is also using memberlist / port 7946
+        #- name: METALLB_ML_BIND_PORT
+        #  value: "7946"
+        # - name: METALLB_ML_LABELS
+        #   value: "app=metallb,component=speaker"
+        # - name: METALLB_ML_NAMESPACE
+        #   valueFrom:
+        #     fieldRef:
+        #       fieldPath: metadata.namespace
+        # - name: METALLB_ML_SECRET_KEY
+        #   valueFrom:
+        #     secretKeyRef:
+        #       name: memberlist
+        #       key: secretkey
+        image: {{ index .Values.images "metallb-speaker" }}
+        imagePullPolicy: Always
```

Comment:
These are from upstream: https://github.com/metallb/metallb/blob/949fdbd8b873f0fe83a3a7ae7c627bd25dea1592/manifests/metallb.yaml#L317

I guess it could make sense to set this field to `IfNotPresent` upstream, however do you think it makes sense to override this here? In any case, the images are rather small (around 30 MB).

Reply:
The only valid use-case for setting it to `Always` would be if the image tag gets overridden, or? I hope they are not doing that. Probably it's just a left-over and can be corrected in the upstream repo as well.

Reply:
You are probably right. Yes, we don't re-push released images (I'm a MetalLB maintainer). I'll open an issue for this upstream and override here with a comment so that we can merge.

Reply:
Ah, didn't know, very cool! Great :)

Reply:
Actually we have an open issue: https://github.com/metallb/metallb/issues/748
Thanks for catching this :+1: